lib/stats/DefaultStatsFactoryPlugin.js

```diff
@@ -351,14 +603,16 @@ const SIMPLE_EXTRACTORS = {
 					array.push(chunk);
 				}
 			}
-			/** @type {Map<string, ExtendedAsset>} */
+			/** @type {Map<string, StatsAsset>} */
 			const assetMap = new Map();
+			/** @type {Set<StatsAsset>} */
 			const assets = new Set();
 			for (const asset of compilation.getAssets()) {
+				/** @type {StatsAsset} */
 				const item = {
 					...asset,
 					type: "asset",
-					related: undefined
+					related: []
```

Comment:
When testing locally, when this is `undefined`, the serialization converts this to an object, but from what I can tell in this section, we are expecting an array. Not sure if we should be tampering with the output generator or correcting it here?